src/transformers/models/kosmos2_5/processing_kosmos2_5.py

```diff
@@ -0,0 +1,154 @@
+
+    def __call__(
+        self,
+        images=None,
+        text: Union[TextInput, List[TextInput]] = None,
+        padding: Union[bool, str, PaddingStrategy] = True,
+        truncation: Union[bool, str, TruncationStrategy] = True,
+        max_length: Optional[int] = None,
+        max_patches: Optional[int] = 4096,
+        num_image_tokens: Optional[int] = 2048,
+        stride: int = 0,
+        pad_to_multiple_of: Optional[int] = None,
+        return_attention_mask: Optional[bool] = None,
+        return_tensors: Optional[Union[str, TensorType]] = "pt",
+        **kwargs,
+    ) -> BatchFeature:
+        """
+        This method uses [`Kosmos2_5ImageProcessor.preprocess`] method to prepare image(s) for the model, and
+        [`PreTrainedTokenizerFast.__call__`] to prepare text for the model.
+
+        Please refer to the docstring of the above two methods for more information.
+
+        The rest of this documentation shows the arguments specific to `Kosmos2_5Processor`.
+        """
+        if images is None and text is None:
+            raise ValueError("You have to specify either images or text.")
+
+        if images is None:
+            raise ValueError("Kosmos2_5Processor requires images to be passed.")
+
+        encoding = BatchFeature()
+
+        if images is not None:
+            image_encoding = self.image_processor(
+                images, return_tensors=return_tensors, max_patches=max_patches, **kwargs
+            )
+            image_encoding.pop("rows")
+            image_encoding.pop("cols")
+            encoding.update(image_encoding)
+
+        if text is not None:
+            # use updates or pop
+            input = self.tokenizer(
+                text,
+                padding=padding,
+                truncation=truncation,
+                max_length=max_length,
+                stride=stride,
+                pad_to_multiple_of=pad_to_multiple_of,
+                return_attention_mask=return_attention_mask,
+                return_tensors="pt",
+            )
+
+            batch_size, seq_len = input.input_ids.shape
+
+            additional_tokens = [self.bos, self.boi] + [self.bos] * num_image_tokens + [self.eoi]
+            additional_tokens_tensor = torch.tensor(additional_tokens).unsqueeze(0).repeat(batch_size, 1)
+            input_ids = torch.cat([additional_tokens_tensor, input.input_ids], dim=1)
+
+            # 1 is image
+            image_embeds_position_mask = [0, -1] + [1] * num_image_tokens + [-1] + [0] * seq_len
+            image_embeds_position_mask = (
+                torch.LongTensor(image_embeds_position_mask).unsqueeze(0).repeat(batch_size, 1)
+            )
+
+            added_attention_mask = [1, 1] + [1] * num_image_tokens + [1]
+            added_attention_mask_tensor = torch.tensor(added_attention_mask).unsqueeze(0).repeat(batch_size, 1)
+            attention_mask = torch.cat([added_attention_mask_tensor, input.attention_mask], dim=1)
```

Comment:
why don't you simply tokenize with the tokens added in string format, it would simplify quite a lot no?

Reply:
changed to string now